### PR TITLE
Refactor GasTracer

### PR DESCRIPTION
**Pull request by s2quake**
### Changes

* Added Added `CancelTrace` method to `GasTracer`.

### CODE REVIEW

File: src/Libplanet.Action/ActionEvaluator.cs [hunk 1]
```diff
@@ -285,6 +286,7 @@ IActionContext CreateActionContext(IWorld newPrevState)
                     randomSeed: inputContext.RandomSeed,
                     isPolicyAction: isPolicyAction,
                     maxGasPrice: tx?.MaxGasPrice,
+                    gasLimit: tx?.GasLimit,
```
Comment (riemannulus) [inline src/Libplanet.Action/ActionEvaluator.cs:289]: Is it needed?

File: src/Libplanet.Action/GasMeter.cs [hunk 2]
```diff
@@ -2,13 +2,18 @@ namespace Libplanet.Action
 {
     internal class GasMeter : IGasMeter
     {
-        public GasMeter(long gasLimit, long gasUsed = 0)
+        public GasMeter(long gasLimit, long gasAvailable)
```
Comment (riemannulus) [inline src/Libplanet.Action/GasMeter.cs:5]: There is a huge difference between tracking usage and tracking available. Could you explain about refactoring?

File: src/Libplanet.Action/ActionEvaluator.cs (hunk 1 above)
Comment (s2quake) [reply src/Libplanet.Action/ActionEvaluator.cs]: > Is it needed?

That code has been reverted.

**Review by riemannulus — DISMISSED** [review]
LGTM! But, Could you make the test about `GasTracer.IsTraceCancelled`?

**Comment by s2quake** [comment]
> LGTM! But, Could you make the test about `GasTracer.IsTraceCancelled`?

Test code for `GasTracer.IsTraceCancelled` has been committed. https://github.com/planetarium/libplanet/pull/3974/commits/6d4da2aaf8dcafe1f4774421d92c33e4c0ce1b24

**Review by riemannulus — APPROVED** [review]

**Pull request merged**